### ISSUE TITLE
PIVOT accepts an ORDER BY clause which is ignored

### ISSUE THREAD

**Issue by paultiq** [duckdb/duckdb#7124]
### What happens?

PIVOT accepts an ORDER BY clause which is ignored

`PIVOT Cities ON Year USING SUM(Population) GROUP BY country ORDER BY country desc`
`PIVOT Cities ON Year USING SUM(Population) GROUP BY country ORDER BY country desc`

return identical results: the "order by" is ignored. It's not in the syntax on https://duckdb.org/docs/sql/statements/pivot.html, but I wrote this out of habit: an "order by" following a "group by" is common practice. 

I'd be happy with a syntax error.

### To Reproduce

```
CREATE TABLE Cities(Country VARCHAR, Name VARCHAR, Year INT, Population INT);
INSERT INTO Cities VALUES ('NL', 'Amsterdam', 2000, 1005);
INSERT INTO Cities VALUES ('NL', 'Amsterdam', 2010, 1065);
INSERT INTO Cities VALUES ('NL', 'Amsterdam', 2020, 1158);
INSERT INTO Cities VALUES ('US', 'Seattle', 2000, 564);
INSERT INTO Cities VALUES ('US', 'Seattle', 2010, 608);
INSERT INTO Cities VALUES ('US', 'Seattle', 2020, 738);
INSERT INTO Cities VALUES ('US', 'New York City', 2000, 8015);
INSERT INTO Cities VALUES ('US', 'New York City', 2010, 8175);
INSERT INTO Cities VALUES ('US', 'New York City', 2020, 8772);
FROM Cities;
PIVOT Cities ON Year USING SUM(Population) GROUP BY country ORDER BY country desc
```

### OS:

Windows

### DuckDB Version:

0.7.2.dev1734

### DuckDB Client:

Python

### Full Name:

Paul Timmins

### Affiliation:

Iqmo

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mytherin** [comment]
Thanks for the report! This should be fixed in #7163

**Comment by paultiq** [comment]
thanks!

If order and limit do work, then I'd also suggest a doc update to the syntax on https://duckdb.org/docs/sql/statements/pivot.html
